src/components/ActivityForm.js

```diff
@@ -27,57 +27,68 @@ export default class ActivityForm extends React.Component {
       name: props.activity ? props.activity.name : '',
       room: props.activity ? props.activity.room : '',
       teacher: props.activity ? props.activity.teacher : '',
-      error: '',
+      classNoError: '',
+      nameError: '',
+      roomError: '',
+      teacherError: '',
       // TODO: when you will use prop-types you will be able to improve this place ;)
       // TODO: I think, that now (looking at previous comments, maybe you will be able to
       // improve). If not, because activity will be able to be undefined, in separate PR
       // you should check lodash package and get method ;)
     };
   }
 
-  onNameChange = e => this.setState({ name: e.target.value });
+  onNameChange = e => {
+    this.setState({ name: e.target.value });
+    this.setState({ nameError: '' });
```

Comment:
at this moment, app will be rerendered twice on each onNameChange...
It is very bad for performance. You can simply make:
```
this.setState({
  name: e.target.value,
  nameError: '',
})
```

Reply:
of, it should be applied to all occurances